web/modules/custom/custom_migration_users/src/Plugin/migrate/destination/MyCustomFields.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Drupal\custom_migration_users\Plugin\migrate\destination;
+
+use Drupal\migrate\Annotation\MigrateDestination;
+use Drupal\migrate\Plugin\MigrateDestinationInterface;
+use Drupal\migrate\Row;
+
+/**
+ * Custom migrate destination plugin for users extra fields.
+ *
+ * @MigrateDestination(
+ *   id = "my_custom_fields"
+ * )
+ */
+class MyCustomFields implements MigrateDestinationInterface {
+
+  /**
+   * {@inheritDoc}
+   */
+  public function getIds(): array {
+    return ['uid' => ['type' => 'integer']];
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function fields(): array {
+    return [
+      ['uid' => ['type' => 'integer']],
+      ['city_name' => ['type' => 'string']],
+      ['country' => ['type' => 'string']],
+      ['interested' => ['type' => 'string']],
+    ];
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function import(Row $row, array $old_destination_id_values = []): void {
+    $uid = $row->getSourceProperty('uid');
+    $country = $row->getSourceProperty('country');
+    $city_name = $row->getSourceProperty('city_name');
+    $interested = $row->getSourceProperty('interested');
```

Comment:
There are 2 "groups" of data
Source and Destination
Source is the data as it was returned from the source plugin
Destination is data after processing
Here you ignoring processing, and it works just because your process is doing nothing